### DISCUSSION TITLE
IDP - Consent user to save data before create user

### DISCUSSION THREAD

**Discussion by mrharsalla** [keycloak/keycloak#36176] [Q&A]
Hi,
I tried to modify the login flow to first show consent to the user before saving their data. But, the form displays the mandatory fields to fill in. How can I force the user to accept/refuse consent before saving their data?
Thanks,

**Comment by cgeorgilakis** [comment]
Unfortunatelly, Keycloak does not follow GDPR for broker users as you mention.
I have submitted a [PR](https://github.com/keycloak/keycloak/pull/29975) for solving this issue.
